### PR TITLE
Test reversion with custom errors & maximum view call size

### PR DESCRIPTION
**Pull request by CedarMist**

### CODE REVIEW

**Comment by CedarMist** [comment]
This has been tested on sapphire-testnet & locally with an un-borked version of sapphire-dev.

Currently it fails on testnet without https://github.com/oasisprotocol/oasis-web3-gateway/pull/468

File: contracts/test/semantics.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    const f = (await ethers.getContractFactory(
+      'SemanticTests',
+    )) as SemanticTests__factory;
+    c = await f.deploy();
+    await c.deployed();
+    chainId = (await c.provider.getNetwork()).chainId;
+  });
+
+  it('eth_call maximum return length vs gas limit', async () => {
+    if (chainId != 31337) {
```
Comment (matevz) [inline contracts/test/semantics.ts:25]: I would expect sapphire-contracts tests not to run on hardhat node. On the CI, do we test it against hardhat node too?

File: contracts/hardhat.config.ts [hunk 2]
```diff
@@ -64,7 +64,7 @@ const config: HardhatUserConfig = {
       chainId: 0x5aff,
       accounts: process.env.SAPPHIRE_TESTNET_PRIVATE_KEY
         ? [process.env.SAPPHIRE_TESTNET_PRIVATE_KEY]
-        : [],
+        : TEST_HDWALLET,
```
Comment (matevz) [inline contracts/hardhat.config.ts:67]: Is this deliberate or a leftover?

**Review by matevz — APPROVED** [review]
Some minor nits, othwerwise lgtm.

File: contracts/hardhat.config.ts [hunk 3]
```diff
@@ -64,7 +64,7 @@ const config: HardhatUserConfig = {
       chainId: 0x5aff,
       accounts: process.env.SAPPHIRE_TESTNET_PRIVATE_KEY
         ? [process.env.SAPPHIRE_TESTNET_PRIVATE_KEY]
-        : TEST_HDWALLET,
+        : [],
```
Comment (matevz) [inline contracts/hardhat.config.ts:67]: Here we go again :smile:

**Review by matevz — APPROVED** [review]

File: contracts/hardhat.config.ts (hunk 3 above)
Comment (CedarMist) [reply contracts/hardhat.config.ts:67]: So?

Comment (matevz) [reply contracts/hardhat.config.ts:67]: Just revert it?

```suggestion
        : TEST_HDWALLET,
```

**Pull request merged**